pkg/prometheus/server/statefulset.go

```diff
@@ -690,17 +691,54 @@ func createThanosContainer(p *monitoringv1.Prometheus, c prompkg.Config) (*v1.Co
 	if thanos.GetConfigInterval != "" && thanosVersion.GTE(semver.MustParse("0.29.0")) {
 		thanosArgs = append(thanosArgs, monitoringv1.Argument{Name: "prometheus.get_config_interval", Value: string(thanos.GetConfigInterval)})
 	}
-	if thanosVersion.GTE(semver.MustParse(thanosSupportedVersionHTTPClientFlag)) {
-		thanosArgs = append(thanosArgs, monitoringv1.Argument{Name: "prometheus.http-client", Value: `{"tls_config": {"insecure_skip_verify":true}}`})
+
+	args, vs, vms, err := thanosBuildVolumes(p, thanosVersion)
```

Comment:
(suggestion) I would avoid the intermediate thanosBuildVolumes() function and just inline the code here

````
if thanosVersion.GTE(semver.MustParse(thanosSupportedVersionHTTPClientFlag)) {
    thanosArgs = append(thanosArgs, ...)
    container.VolumeMounts = append(container.VolumeMounts, ...)
    ...
````